fairscale/nn/pipe/async_pipe.py

```diff
@@ -192,6 +192,8 @@ def __init__(
             warnings.warn("More ranks than partitions, some ranks unused")
             self.partitions: List[ModuleWrapper] = []
             self.pipeline = None
+            # TODO(msb) remove this hack
```

Comment:
Can you give some context about what it should be instead?

Reply:
AsyncPipe seems to manage partitions as a list of ModuleWrapper while MultiProcessPipe only manages a single nn.Sequential. I don't totally understand what the List[ModuleWrapper] logic is all about and there don't seem to be any tests for that really. All the tests just inspect the first partition in the list. So this hack allows to simply MultiProcessPipe while allowing the tests to continue functioning.

I'm currently focused on rationalizing MultiProcessPipe. When I move on to AsyncPipe, I'm plan to fully understand what is actually going on here. I don't totally know how to undo this hack just yet. My hope is that I can remove List[ModuleWrapper] and just have a single nn.Sequential. But I currently don't know if that is possible.